### PR TITLE
CIP-64: New Transaction Type: Celo Dynamic Fee v2

### PR DESCRIPTION
**Pull request by gastonponti**

### CODE REVIEW

File: CIPs/cip-0064.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+status: Accepted
+type: Standards Track
+category: Ring 0
+created: 2023-07-10
+license: Apache 2.0
+---
+
+## Simple Summary
+
+Create a new [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) Transaction Type that will act as [eip1559](https://eips.ethereum.org/EIPS/eip-1559) tx but including the `feeCurrency` field for paying in a different currency.
```
Comment (palango) [inline CIPs/cip-0064.md:15]: Please spell out *transaction * here.

File: CIPs/cip-0064.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Abstract
+
+This CIP proposes to create a new EIP-2718 Transaction Type `0x7b`, which will have the same functionality as the [CIP-42](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0042.md), but removing the `gateway` fields (already deprecated in the [CIP-57](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0057.md)).
```
Comment (palango) [inline CIPs/cip-0064.md:19]: Please mention both complete field names.

File: CIPs/cip-0064.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+With the format `0x7b || rlp([chainId, nonce, maxPriorityFeePerGas, maxFeePerGas, gasLimit, to, value, data, accessList, feeCurrency, signatureYParity, signatureR, signatureS])`
```
Comment (palango) [inline CIPs/cip-0064.md:20]: This sentences misses some parts. Something like: *This transaction is encoded as follows.*

File: CIPs/cip-0064.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Motivation
+
+In the Espresso Hard Fork the [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) was introduced. We also added a new transaction type `0x7c`, which is the `CeloDynamicFee` transaction from [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559), with the ability to be paid in a different currency. This transaction type was added as part of the [CIP-42](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0042.md).
```
Comment (palango) [inline CIPs/cip-0064.md:24]: No *the* necessary before EIP

File: CIPs/cip-0064.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+The `signatureYParity, signatureR, signatureS` elements of this transaction represent a secp256k1 signature over `keccak256(0x7b || rlp([chainId, nonce, maxPriorityFeePerGas, maxFeePerGas, gasLimit, to, value, data, accessList, feeCurrency]))`.
+
+The [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) `ReceiptPayload` for this transaction is `rlp([status, cumulativeGasUsed, logsBloom, logs])`.
+
+## Rationale
+
+The creation of this new transaction type will maintain functionality that we already have and will also allow us to prune our codebase in the L2 migration. That future migration will require to reduce differences to Ethereum as much as we can to reduce the complexity added for proof creations in either scenario (optimistic and zk), for example the future removal of the transaction added in the CIP-42 plus reverting the celo legacy tx, to be exactly the same as the legacy in Ethereum. Also, it will give us the opportunity to prune our codebase and make it more maintainable.
```
Comment (palango) [inline CIPs/cip-0064.md:46]: please spell out *transaction* instead of *tx*.

**Review by palango — APPROVED** [review]
Some small changes necessary, but with those in this PR is ready to be merged.

**Review by kokobuna7 — COMMENTED** [review]
Pay ID me @ thedoggydogydog53@gmail.com via commonwealth bank account  or revolut me/@Virgilouvn

**Pull request merged**